assets/js/components/Dashboard/PaMessagesPage.tsx

```diff
@@ -1,29 +1,84 @@
-import React, { ComponentType, useState, useEffect } from "react";
-import { Container, Row, Col, FormCheck } from "react-bootstrap";
+import React, { ComponentType, useState, useMemo, useEffect } from "react";
+import {
+  Container,
+  Row,
+  Col,
+  FormCheck,
+  ButtonGroup,
+  Button,
+} from "react-bootstrap";
 import { PlusCircleFill } from "react-bootstrap-icons";
-import { fetchPaMessages } from "Utils/api";
 import { PaMessage } from "Models/pa_message";
-import { Link } from "react-router-dom";
+import { Link, useSearchParams } from "react-router-dom";
+import cx from "classnames";
+import useSWR from "swr";
+
+type State = "active" | "future" | "past";
+
+const fetcher = (...args: Parameters<typeof fetch>) =>
+  fetch(...args).then((resp) => resp.json());
+const usePaMessages = ({ state }: { state?: State | null }) => {
+  const url = useMemo(() => {
+    const params = new URLSearchParams();
+
+    if (state) params.set("state", state);
+
+    return `/api/pa-messages?${params.toString()}`;
+  }, [state]);
+
+  return useSWR(url, fetcher);
+};
 
 const PaMessagesPage: ComponentType = () => {
-  const [paMessages, setPaMessages] = useState<PaMessage[]>([]);
+  const [params, setParams] = useSearchParams();
+  const [state, setState] = useState<State | null>(
+    () => (params.get("state") as State) ?? null,
+  );
 
   useEffect(() => {
-    const fetchAndSetPaMessages = async () => {
-      const allPaMessages = await fetchPaMessages();
-      setPaMessages(allPaMessages);
-    };
-    fetchAndSetPaMessages();
-  }, []);
+    setParams(() => {
+      const newParams = new URLSearchParams();
+
+      if (state) newParams.set("state", state);
```

Comment:
We're building the URL params twice in the exact same way here (L22~ and L40~). Should we de-duplicate so these will be easier to keep in sync?

Reply:
These will diverge when we add filters on service type / route

- https://github.com/mbta/screenplay/pull/431/files#diff-63017d34d99e9b41261c38d09103b8c5eb957968e3ac83d6824c7cfc809ce03fR61-R69

- https://github.com/mbta/screenplay/pull/431/files#diff-63017d34d99e9b41261c38d09103b8c5eb957968e3ac83d6824c7cfc809ce03fR88-R92